api/src/main/java/org/pmiops/workbench/cohortbuilder/SearchGroupItemQueryBuilder.java

```diff
@@ -0,0 +1,578 @@
+    }
+    // Otherwise build sql against flat denormalized search table
+    for (SearchParameter param : searchGroupItem.getSearchParameters()) {
+      if (param.getAttributes().isEmpty()) {
+        childConceptIds.addAll(childConceptIds(criteriaLookup, ImmutableList.of(param)));
+        // make sure we only add the concept ids sql template once
+        if (!queryParts.contains(CONCEPT_ID_IN)) {
+          queryParts.add(CONCEPT_ID_IN);
+        }
+      } else {
+        StringBuilder bpSql = new StringBuilder(BP_SQL);
+        List<Long> bpConceptIds = new ArrayList<>();
+        for (Attribute attribute : param.getAttributes()) {
+          if (attribute.getConceptId() != null) {
+            // this indicates we need to build a blood pressure sql statement
+            bpConceptIds.add(attribute.getConceptId());
+            processBloodPressureSql(queryParams, bpSql, attribute);
+          } else if (AttrName.NUM.equals(attribute.getName())) {
+            queryParts.add(processNumericalSql(queryParams, param, attribute));
+          } else {
+            queryParts.add(processCategoricalSql(queryParams, param, attribute));
+          }
+        }
+        if (!bpConceptIds.isEmpty()) {
+          // if blood pressure we need to add named parameters for concept ids
+          QueryParameterValue cids =
+              QueryParameterValue.array(bpConceptIds.stream().toArray(Long[]::new), Long.class);
+          String conceptIdsParam = addQueryParameterValue(queryParams, cids);
+          queryParts.add(String.format(bpSql.toString(), conceptIdsParam) + ")\n");
+        }
+      }
+      // all search parameters in the search group item have the same source/standard flag
+      isStandard = param.getStandard();
+    }
+    // need to OR all query parts together since they exist in the same search group item
+    String queryPartsSql = "(" + String.join(OR, queryParts) + ")\n";
+    if (!childConceptIds.isEmpty()) {
+      // if we have concept ids that are non bp then add named parameters
+      QueryParameterValue cids =
+          QueryParameterValue.array(childConceptIds.stream().toArray(Long[]::new), Long.class);
+      String conceptIdsParam = addQueryParameterValue(queryParams, cids);
+      queryPartsSql = String.format(queryPartsSql, conceptIdsParam);
+    }
+    // create the source/standard flag named parameter
+    String standardParam =
+        addQueryParameterValue(queryParams, QueryParameterValue.int64(isStandard ? 1 : 0));
+    // format the base sql with all query parts
+    String baseSql = String.format(BASE_SQL, standardParam) + queryPartsSql;
+    // build modifier sql if modifiers exists
+    String modifiedSql = buildModifierSql(baseSql, queryParams, searchGroupItem.getModifiers());
+    // split off the sql conditions in case were building a temporal sql statement
+    // otherwise this is just ignored
+    String conditionsSql = baseSql.split("where")[1] + "%s";
```

Comment:
This is a bit tough to follow and verify - perhaps the conditions clause can be constructed separately? Seems difficult to verify that there's exactly one WHERE clause in the sql query.

Reply:
Fixed.